src/cpp/rtps/participant/RTPSParticipantImpl.cpp

```diff
@@ -103,10 +105,54 @@ RTPSParticipantImpl::RTPSParticipantImpl(const RTPSParticipantAttributes& PParam
         m_network_Factory.RegisterTransport(&descriptor);
     }
 
-    // User defined transports
-    for (const auto& transportDescriptor : PParam.userTransports)
+    /// Creation of metatraffic locator and receiver resources
+    uint32_t metatraffic_multicast_port = m_att.port.getMulticastPort(m_att.builtin.domainId);
```

Comment:
After the explanation of why these changes were done, I think the user is responsible for configuring participants that use TCP transport.

Reply:
automatic generation of TCP listener ports for clients is now removed